hledger-lib/Hledger/Data/AutoTransaction.hs

```diff
@@ -146,6 +146,53 @@ renderPostingCommentDates p = p { pcomment = comment' }
 -- 2017/03/01
 --     hi           $1.00
 -- <BLANKLINE>
+-- >>> mapM_ (putStr . show) $ runPeriodicTransaction (PeriodicTransaction "monthly from 2017/1 to 2017/5" ["hi" `post` usd 1]) (mkdatespan "2017/02/01" "2017/04/02")
+-- 2017/02/01
+--     hi           $1.00
+-- <BLANKLINE>
+-- 2017/03/01
+--     hi           $1.00
+-- <BLANKLINE>
+-- 2017/04/01
+--     hi           $1.00
+-- <BLANKLINE>
+-- >>> mapM_ (putStr . show) $ runPeriodicTransaction (PeriodicTransaction "every 2nd day of month" ["hi" `post` usd 1]) (mkdatespan "2017/02/01" "2017/04/02")
+-- 2017/02/02
+--     hi           $1.00
+-- <BLANKLINE>
+-- 2017/03/02
+--     hi           $1.00
+-- <BLANKLINE>
+-- >>> mapM_ (putStr . show) $ runPeriodicTransaction (PeriodicTransaction "monthly from 2017/1/5 to 2017/4" ["hi" `post` usd 1]) nulldatespan
+-- 2017/01/01
+--     hi           $1.00
+-- <BLANKLINE>
+-- 2017/02/01
+--     hi           $1.00
+-- <BLANKLINE>
+-- 2017/03/01
+--     hi           $1.00
+-- <BLANKLINE>
+-- >>> mapM_ (putStr . show) $ runPeriodicTransaction (PeriodicTransaction "every 30rd day of month from 2017/1 to 2017/5" ["hi" `post` usd 1]) nulldatespan
+-- 2017/01/30
+--     hi           $1.00
+-- <BLANKLINE>
+-- 2017/03/30
```

Comment:
I suspect it's better to adjust to the last day of shorter months (eg 2/28 or 2/29), rather than skipping the month ?

Reply:
Makes sense. I followed an idea that if we've asked to have it `every 30rd day of the month from 2017/2` then we want excactly that day of a month and we aware that not all months have that day. Same with `every 5th tue of the month in 2017`.
Anyway there is still som undecided stuff regarding periods. Not always we want flexible intervals that spans outside of requested `DateSpan`.